### PR TITLE
Document p2p package

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

Adds some docs to the p2p package

## How this works

adds comments

## How this was tested

CI

### CODE REVIEW

File: network/p2p/router.go [hunk 1]
```diff
@@ -198,6 +199,9 @@ func (r *Router) RegisterAppProtocol(handlerID uint64, handler Handler, nodeSamp
 	}, nil
 }
 
+// AppRequest routes an AppRequest to a Handler based on the handler prefix.
+//
+// The message is dropped if no matching handler can be found.
```
Comment (aaronbuchwald) [inline network/p2p/router.go:204]: It's a bit non-obvious here why `handler.AppRequest` propagates an error as fatal if it returns an error.

It looks like the reason for this is that the error returned from the handler gets dropped here: 

https://github.com/ava-labs/avalanchego/blob/master/network/p2p/handler.go#L99

but a fatal error could be returned from the `AppSender` when trying to send the response to the peer here: 

https://github.com/ava-labs/avalanchego/blob/master/network/p2p/handler.go#L111

Could we either add a comment internally to this function explaining why this is the one case where it looks like we're returning a fatal error from the handler or find a different way to make it more clear?

**Review by aaronbuchwald — APPROVED** [review]
Left one comment about making fatal error more obviously correct.

Other than that, this PR LGTM thanks for putting this up!

File: network/p2p/router.go [hunk 2]
```diff
@@ -220,6 +224,10 @@ func (r *Router) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID ui
 	return nil
 }
 
+// AppRequestFailed routes an AppRequestFailed message to the callback
+// corresponding to requestID.
+//
+// A fatal error has occurred if no corresponding requestID can be found.
```
Comment (danlaine) [inline network/p2p/router.go:230]: Maybe we should be more concrete and say that `ErrUnrequestedResponse` should be considered fatal?

**Review by danlaine — APPROVED** [review]

**Review by StephenButtolph — COMMENTED** [review]
just one small nit - lgtm

File: network/p2p/router.go (hunk 1 above)
Comment (joshua-kim) [reply network/p2p/router.go:204]: Added a comment

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**